src/main/java/entities/CollaborativeTask.java

```diff
@@ -0,0 +1,261 @@
+     */
+    public void removePendingTeammates(ArrayList<StudentUser> oldTeammates){
+        this.pendingTeammates.removeAll(oldTeammates);
+    }
+
+    /**
+     * @return an Array List of all Student Users who have declined invitations to join the Collaborative Task.
+     */
+    public ArrayList<StudentUser> getDeclinedTeammates(){
+        return this.declinedTeammates;
+    }
+
+    /**
+     * To be used when invitations to join a Collaborative Task are declined.
+     * @param newTeammates - an Array List of Student Users who have declined invitations to join the Collaborative Task.
+     */
+    public void addDeclinedTeammates(ArrayList<StudentUser> newTeammates){
+        this.declinedTeammates.addAll(newTeammates);
+    }
+
+    /**
+     * To be used when a Student User who has previously declined an invitation to join the Collaborative Task has
+     * accepted a reinvitation to join.
+     * @param oldTeammates - an Array List of Student Users who have accepted an invitation to join the Collaborative
+     *                       Task after previously declining such an invitation.
+     */
+    public void removeDeclinedTeammates(ArrayList<StudentUser> oldTeammates){
+        this.declinedTeammates.removeAll(oldTeammates);
+    }
+
+    /**
+     * @return whether the Collaborative Task is reoccurring or not.
+     */
+    public boolean getRecurring(){
+        return this.recurring;
+    }
+
+    /**
+     * @return frequency of Collaborative Task.
+     */
+    public String getFrequency(){
+        return this.frequency;
+    }
+
+    /**
+     * Set a Collaborative Task as recurring/not
+     * Set its new frequency if it is recurring
+     * @param recurring - whether the Collaborative Task is recurring
+     * @param frequency - the frequency at which the Collaborative Task occurs (if recurring)
+     */
+    public void setRecurring(boolean recurring, String frequency) {
+        this.recurring = recurring;
+        if (recurring) {
+            this.frequency = frequency;
+        }
+        else{
+            this.frequency = "";
+        }
+    }
+
+    /**
+     * @return deadline for Collaborative Task.
+     */
+    public LocalDateTime getDeadline(){
+        return this.deadline;
+    }
+
+    /**
+     * Set a deadline for a Collaborative Task.
+     * @param deadline - deadline for Collaborative Task.
+     */
+    public void setDeadline(LocalDateTime deadline){
+       this.deadline = deadline;
+    }
+
+    /**
+     * Get the time blocks of a Collaborative Task.
+     * @return - the time blocks of the Collaborative task in an Array List of Array Losts of LocalDateTimes.
+     */
+    public ArrayList<ArrayList<LocalDateTime>> getTimeBlocks() {
+        return this.timeBlocks;
+    }
+
+    /**
+     * Set new time blocks
+     * @param timeBlocks - Array List of Array Lists of time blocks of the Collaborative Task.
+     */
+    public void setTimeBlocks(ArrayList<ArrayList<LocalDateTime>> timeBlocks) {
+
```

Comment:
I think you forgot part of the code for setTimeBlocks (i.e. this.timeBlocks = timeBlocks)